packages/yarnpkg-core/sources/Manifest.ts

```diff
@@ -356,6 +356,9 @@ export class Manifest {
         continue;
       }
 
+      if (this.workspaceDefinitions.some(({pattern}) => pattern === entry))
+        continue;
+
```

Comment:
Could we use a Set with the patterns in it to make the lookup faster?

Reply:
Makes sense.

I've added the `Set` to the `load` function to keep the API of `Manifest` as is.